### ISSUE TITLE
v1.0.0 installation failed

### ISSUE THREAD

**Issue by drcege** [HazyResearch/flash-attention#163]
Several days ago, I can successfully install flash-attn by `pip install flash-attn`. However today, the same command failed. After testing, I found that `pip install flash-attn==0.2.8` still works.

Here is the stack trace:
```bash
[1/9] /usr/local/cuda/bin/nvcc  -I/tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn -I/tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src -I/tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/cutlass/include -I/usr/local/lib/python3.8/dist-packages/torch/include -I/usr/local/lib/python3.8/dist-packages/torch/include/torch/csrc/api/include -I/usr/local/lib/python3.8/dist-packages/torch/include/TH -I/usr/local/lib/python3.8/dist-packages/torch/include/THC -I/usr/local/cuda/include -I/usr/include/python3.8 -c -c /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha_bwd_hdim64.cu -o /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/build/temp.linux-x86_64-3.8/csrc/flash_attn/src/fmha_bwd_hdim64.o -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr --compiler-options ''"'"'-fPIC'"'"'' -O3 -std=c++17 -U__CUDA_NO_HALF_OPERATORS__ -U__CUDA_NO_HALF_CONVERSIONS__ -U__CUDA_NO_HALF2_OPERATORS__ -U__CUDA_NO_BFLOAT16_CONVERSIONS__ --expt-relaxed-constexpr --expt-extended-lambda --use_fast_math --ptxas-options=-v -lineinfo -gencode arch=compute_75,code=sm_75 -gencode arch=compute_80,code=sm_80 -gencode arch=compute_90,code=sm_90 --threads 4 -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1013"' -DTORCH_EXTENSION_NAME=flash_attn_cuda -D_GLIBCXX_USE_CXX11_ABI=1
      FAILED: /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/build/temp.linux-x86_64-3.8/csrc/flash_attn/src/fmha_bwd_hdim64.o
      /usr/local/cuda/bin/nvcc  -I/tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn -I/tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src -I/tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/cutlass/include -I/usr/local/lib/python3.8/dist-packages/torch/include -I/usr/local/lib/python3.8/dist-packages/torch/include/torch/csrc/api/include -I/usr/local/lib/python3.8/dist-packages/torch/include/TH -I/usr/local/lib/python3.8/dist-packages/torch/include/THC -I/usr/local/cuda/include -I/usr/include/python3.8 -c -c /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha_bwd_hdim64.cu -o /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/build/temp.linux-x86_64-3.8/csrc/flash_attn/src/fmha_bwd_hdim64.o -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr --compiler-options ''"'"'-fPIC'"'"'' -O3 -std=c++17 -U__CUDA_NO_HALF_OPERATORS__ -U__CUDA_NO_HALF_CONVERSIONS__ -U__CUDA_NO_HALF2_OPERATORS__ -U__CUDA_NO_BFLOAT16_CONVERSIONS__ --expt-relaxed-constexpr --expt-extended-lambda --use_fast_math --ptxas-options=-v -lineinfo -gencode arch=compute_75,code=sm_75 -gencode arch=compute_80,code=sm_80 -gencode arch=compute_90,code=sm_90 --threads 4 -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1013"' -DTORCH_EXTENSION_NAME=flash_attn_cuda -D_GLIBCXX_USE_CXX11_ABI=1
      In file included from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/cutlass/include/cutlass/arch/memory_sm75.h:38,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/cutlass/include/cutlass/gemm/warp/mma_tensor_op.h:46,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/cutlass/include/cutlass/gemm/warp/default_mma_tensor_op.h:38,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha/gemm.h:33,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha/smem_tile.h:32,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha_kernel.h:34,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha_fprop_kernel_1xN.h:31,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha_dgrad_kernel_1xN_loop.h:6,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha_bwd_launch_template.h:7,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha_bwd_hdim64.cu:5:
      /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/cutlass/include/cutlass/layout/matrix.h:42:10: fatal error: cute/layout.hpp: No such file or directory
         42 | #include "cute/layout.hpp"
            |          ^~~~~~~~~~~~~~~~~
      compilation terminated.
[2/9] /usr/local/cuda/bin/nvcc  -I/tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn -I/tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src -I/tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/cutlass/include -I/usr/local/lib/python3.8/dist-packages/torch/include -I/usr/local/lib/python3.8/dist-packages/torch/include/torch/csrc/api/include -I/usr/local/lib/python3.8/dist-packages/torch/include/TH -I/usr/local/lib/python3.8/dist-packages/torch/include/THC -I/usr/local/cuda/include -I/usr/include/python3.8 -c -c /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha_fwd_hdim64.cu -o /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/build/temp.linux-x86_64-3.8/csrc/flash_attn/src/fmha_fwd_hdim64.o -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr --compiler-options ''"'"'-fPIC'"'"'' -O3 -std=c++17 -U__CUDA_NO_HALF_OPERATORS__ -U__CUDA_NO_HALF_CONVERSIONS__ -U__CUDA_NO_HALF2_OPERATORS__ -U__CUDA_NO_BFLOAT16_CONVERSIONS__ --expt-relaxed-constexpr --expt-extended-lambda --use_fast_math --ptxas-options=-v -lineinfo -gencode arch=compute_75,code=sm_75 -gencode arch=compute_80,code=sm_80 -gencode arch=compute_90,code=sm_90 --threads 4 -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1013"' -DTORCH_EXTENSION_NAME=flash_attn_cuda -D_GLIBCXX_USE_CXX11_ABI=1
      FAILED: /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/build/temp.linux-x86_64-3.8/csrc/flash_attn/src/fmha_fwd_hdim64.o
      /usr/local/cuda/bin/nvcc  -I/tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn -I/tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src -I/tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/cutlass/include -I/usr/local/lib/python3.8/dist-packages/torch/include -I/usr/local/lib/python3.8/dist-packages/torch/include/torch/csrc/api/include -I/usr/local/lib/python3.8/dist-packages/torch/include/TH -I/usr/local/lib/python3.8/dist-packages/torch/include/THC -I/usr/local/cuda/include -I/usr/include/python3.8 -c -c /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha_fwd_hdim64.cu -o /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/build/temp.linux-x86_64-3.8/csrc/flash_attn/src/fmha_fwd_hdim64.o -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__ -D__CUDA_NO_BFLOAT16_CONVERSIONS__ -D__CUDA_NO_HALF2_OPERATORS__ --expt-relaxed-constexpr --compiler-options ''"'"'-fPIC'"'"'' -O3 -std=c++17 -U__CUDA_NO_HALF_OPERATORS__ -U__CUDA_NO_HALF_CONVERSIONS__ -U__CUDA_NO_HALF2_OPERATORS__ -U__CUDA_NO_BFLOAT16_CONVERSIONS__ --expt-relaxed-constexpr --expt-extended-lambda --use_fast_math --ptxas-options=-v -lineinfo -gencode arch=compute_75,code=sm_75 -gencode arch=compute_80,code=sm_80 -gencode arch=compute_90,code=sm_90 --threads 4 -DTORCH_API_INCLUDE_EXTENSION_H '-DPYBIND11_COMPILER_TYPE="_gcc"' '-DPYBIND11_STDLIB="_libstdcpp"' '-DPYBIND11_BUILD_ABI="_cxxabi1013"' -DTORCH_EXTENSION_NAME=flash_attn_cuda -D_GLIBCXX_USE_CXX11_ABI=1
      In file included from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/cutlass/include/cutlass/arch/memory_sm75.h:38,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/cutlass/include/cutlass/gemm/warp/mma_tensor_op.h:46,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/cutlass/include/cutlass/gemm/warp/default_mma_tensor_op.h:38,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha/gemm.h:33,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha/smem_tile.h:32,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha_kernel.h:34,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha_fprop_kernel_1xN.h:31,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha_fwd_launch_template.h:12,
                       from /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/src/fmha_fwd_hdim64.cu:5:
      /tmp/pip-install-ke2qfajb/flash-attn_e6b14a8f93364de0bf04a59f5df99793/csrc/flash_attn/cutlass/include/cutlass/layout/matrix.h:42:10: fatal error: cute/layout.hpp: No such file or directory
         42 | #include "cute/layout.hpp"
            |          ^~~~~~~~~~~~~~~~~
      compilation terminated.
      
...
and so on
```

**Comment by ZhiyuanChen** [comment]
+1

**Comment by tridao** [comment]
Thanks for the report. I mixed up the cutlass versions when uploading to PyPI. Should be fixed in v1.0.1.
Let me know if that still doesn't work.

**Comment by AlpinDale** [comment]
> Thanks for the report. I mixed up the cutlass versions when uploading to PyPI. Should be fixed in v1.0.1. Let me know if that still doesn't work.

Hi, still doesn't work.

**Comment by tridao** [comment]
> Hi, still doesn't work.

Do you run into the same error, or a different one?
I've just tried a fresh installation `pip install flash-attn` on the nvidia pytorch container `nvcr.io/nvidia/pytorch:22.12-py3` and it works fine.
